packages/netlify-cms-core/src/actions/editorialWorkflow.js

```diff
@@ -230,30 +234,55 @@ function unpublishedEntryDeleteError(collection, slug, transactionID) {
  */
 
 export function loadUnpublishedEntry(collection, slug) {
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
     const state = getState();
     const backend = currentBackend(state.config);
+
     dispatch(unpublishedEntryLoading(collection, slug));
-    backend
-      .unpublishedEntry(collection, slug)
-      .then(entry => dispatch(unpublishedEntryLoaded(collection, entry)))
-      .catch(error => {
-        if (error.name === EDITORIAL_WORKFLOW_ERROR && error.notUnderEditorialWorkflow) {
-          dispatch(unpublishedEntryRedirected(collection, slug));
-          dispatch(loadEntry(collection, slug));
-        } else {
-          dispatch(
-            notifSend({
-              message: {
-                key: 'ui.toast.onFailToLoadEntries',
-                details: error,
-              },
-              kind: 'danger',
-              dismissAfter: 8000,
-            }),
-          );
-        }
-      });
+
+    try {
+      const entry = await backend.unpublishedEntry(collection, slug);
+      const mediaFiles = entry.mediaFiles;
+      const assetProxies = await Promise.all(
+        mediaFiles.map(({ file }) => createAssetProxy(file.name, file)),
+      );
+      dispatch(addAssets(assetProxies));
```

Comment:
sync assets, draft media files and media library on entry load

Reply:
Doing this all in one place is really good for visibility, love it